process/smartContract/process.go

```diff
@@ -1550,6 +1579,11 @@ func (sc *scProcessor) penalizeUserIfNeeded(
 		return
 	}
 
+	// this flag is needed for the transaction simulator in order can calculate the cost of a transaction
+	if sc.bypassChecks {
```

Comment:
rename this to ByPassTooMuchGasCheck

Reply:
I do not like this bypass. in tx simulate you could simply verify the return message and compute from there the actual gasUsed. Please delete the bypass flag